mmv1/products/integrations/AuthConfig.yaml

```diff
@@ -0,0 +1,470 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'AuthConfig'
+description: |
+  The AuthConfig resource use to hold channels and connection config data.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/application-integration/docs/overview'
+    'Manage authentication profiles': 'https://cloud.google.com/application-integration/docs/configure-authentication-profiles'
+  api: 'https://cloud.google.com/application-integration/docs/reference/rest/v1/projects.locations.authConfigs'
+base_url: 'projects/{{project}}/locations/{{location}}/authConfigs'
+self_link: '{{name}}'
+update_verb: :PATCH
```

Comment:
We'll want to add handwritten update test(s) to cover most updatable fields, and all interesting cases (such as clientCertificate).

Reply:
Added update test